// ajv-dates/build/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.times = exports.default = exports.dates = void 0;

var _isAfter = require("./keywords/isAfter");

var _isWeekday = require("./keywords/isWeekday");

var _isWeekend = require("./keywords/isWeekend");

var _isBefore = require("./keywords/isBefore");

var _isBetween = require("./keywords/isBetween");

var _isTimeBetween = require("./keywords/isTimeBetween");

var _isTimeAfter = require("./keywords/isTimeAfter");

var _isTimeBefore = require("./keywords/isTimeBefore");

const dateKeywords = {
  isAfter: _isAfter.isAfter,
  isWeekday: _isWeekday.isWeekday,
  isWeekend: _isWeekend.isWeekend,
  isBefore: _isBefore.isBefore,
  isBetween: _isBetween.isBetween
};
const defaultOptions = {
  parser: val => new Date(val)
};

const dates = function (instance) {
  let options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultOptions;
  return Object.entries(dateKeywords).reduce((acc, _ref) => {
    var _options$keywordMap;

    let [keyword, def] = _ref;
    return acc.addKeyword({
      keyword: ((_options$keywordMap = options.keywordMap) === null || _options$keywordMap === void 0 ? void 0 : _options$keywordMap[keyword]) || keyword,
      ...def({ ...defaultOptions,
        ...options
      })
    }), acc;
  }, instance);
};

exports.dates = dates;
const timeKeywords = {
  isTimeBetween: _isTimeBetween.isTimeBetween,
  isTimeBefore: _isTimeBefore.isTimeBefore,
  isTimeAfter: _isTimeAfter.isTimeAfter
};

const times = (instance, options) => {
  if (!options.parser) throw new Error('A parser must be required');
  return Object.entries(timeKeywords).reduce((acc, _ref2) => {
    var _options$keywordMap2;

    let [keyword, def] = _ref2;
    return acc.addKeyword({
      keyword: ((_options$keywordMap2 = options.keywordMap) === null || _options$keywordMap2 === void 0 ? void 0 : _options$keywordMap2[keyword]) || keyword,
      ...def(options)
    }), acc;
  }, instance);
};

exports.times = times;
var _default = dates;
exports.default = _default;
